### PR TITLE
Feat shirt generator

### PR DESCRIPTION
**Pull request by karol-depka**

### CODE REVIEW

File: InnoTopicWebsite/src/app/cv-page/topics-graph/topics-graph.component.ts [hunk 1]
```diff
@@ -352,6 +352,8 @@ export class TopicsGraphComponent implements OnInit {
       width = +svgRootElement.attr("width"),
       height = +svgRootElement.attr("height");
 
+      svgRootElement.selectAll('*').remove();
```
Comment (karol-depka) [inline InnoTopicWebsite/src/app/cv-page/topics-graph/topics-graph.component.ts:355]: Is this a hack or ?
What is the root cause?

File: InnoTopicWebsite/src/app/shirt-generator/components/preview/preview.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
```
Comment (karol-depka) [inline InnoTopicWebsite/src/app/shirt-generator/components/preview/preview.component.spec.ts:1]: can prolly delete this file.

File: InnoTopicWebsite/src/app/shirt-generator/services/window-dot-ai-topics-prompt.service.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          if (support === 'no') {
+            reject(notSupportedError);
+            return;
+          }
+          const session = await window.ai.createTextSession();
+          try {
+            const result = await session.prompt(text);
+            console.log(`Window AI Prompt Result: `);
+            console.log(result);
+            const splitWords = result.split(',')
```
Comment (karol-depka) [inline InnoTopicWebsite/src/app/shirt-generator/services/window-dot-ai-topics-prompt.service.ts:29]: let's trim them

File: InnoTopicWebsite/src/app/shirt-generator/shirt-generator.page.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  <ion-toolbar>
+    <ion-title>Shirt Generator</ion-title>
+  </ion-toolbar>
+</ion-header>
+
+<ion-content [fullscreen]="true">
+  <app-prompt-input (searchEvent)="onSearch($event)"/>
+    <app-prompt-result
+      [promptResult]="$any(result$ | async)"
+      [isLoading]="$any(isLoading$ | async)"
```
Comment (karol-depka) [inline InnoTopicWebsite/src/app/shirt-generator/shirt-generator.page.html:11]: later we have to take streaming into account ( streaming ends when observable completes )

File: InnoTopicWebsite/src/app/shirt-generator/shirt-generator.page.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  result$ = this.shirtGeneratorService.generatedContent$;
+
+  constructor(
+    private shirtGeneratorService: ShirtGeneratorService,
+  ) {
+  }
+
+  ngOnInit() {
+    this.searchEvent$.pipe(
```
Comment (karol-depka) [inline InnoTopicWebsite/src/app/shirt-generator/shirt-generator.page.ts:25]: I love the pipe functional approach :)

File: InnoTopicWebsite/src/styles.sass [hunk 6]
```diff
@@ -1,169 +1,191 @@
...
+    text-transform: uppercase
+    color: var(--ion-color-secondary)
+    &::after
+      content: ""
+
+  span
+    word-break: break-word
+    width: calc(100% - 55px - 110px)
+
+#languages
```
Comment (karol-depka) [inline InnoTopicWebsite/src/styles.sass:190]: This should prolly be moved to natural-languages components, and I think a random bold is not necessary?

File: InnoTopicWebsite/src/app/cv-page/topics-graph/topics-graph.component.ts (hunk 1 above)
Comment (JayTailor45) [reply InnoTopicWebsite/src/app/cv-page/topics-graph/topics-graph.component.ts:355]: It is not a hack. You have to remove old elements before making the graph. The catch here is that the topics graph was getting loaded twice. 
1. When the application runs for the first time
2. When the application finds that `isPrint` is `true`, and by the time it starts to render it already has data/elements from the first step. Resulting in multiple objects to render.

**Pull request merged**